quinn-proto/src/endpoint.rs

```diff
@@ -369,6 +369,9 @@ impl Endpoint {
         if remote.port() == 0 || remote.ip().is_unspecified() {
             return Err(ConnectError::InvalidRemoteAddress(remote));
         }
+        if !self.config.supported_versions.contains(&config.version) {
```

Comment:
Style nit: I think `else if` would make sense here?

Reply:
Fair enough. I prefer the increased density over syntactic independence, I think. Also in my mind using `else if` would group these more as a set of preconditions, whereas the syntactic independence means I have to do more work parsing in my head because I consider each of them independently.

Anyway, either is fine.

Reply:
(needs reapproval though)